tests/devices/test_default_mixed_tf.py

```diff
@@ -0,0 +1,596 @@
+
+    @pytest.mark.parametrize("c_dtype", [np.complex64, np.complex128])
+    @pytest.mark.parametrize(
+        "measurement",
+        [qml.state(), qml.density_matrix(wires=[1]), qml.density_matrix(wires=[2, 0])],
+    )
+    def test_complex_dtype(self, c_dtype, measurement, tol):
+        """Test that the user-defined dtype of the device is preserved
+        for QNodes with complex-valued outputs"""
+        p = tf.constant(0.543)
+
+        dev = qml.device("default.mixed", wires=3, c_dtype=c_dtype)
+
+        @qml.qnode(dev, interface="tf", diff_method="backprop")
+        def circuit(x):
+            qml.RX(x, wires=0)
+            return qml.apply(measurement)
+
+        res = circuit(p)
+        assert res.dtype == c_dtype
+
+
+class TestOps:
+    """Unit tests for operations supported by the default.qubit.autograd device"""
+
+    def test_multirz_jacobian(self):
+        """Test that the patched numpy functions are used for the MultiRZ
+        operation and the jacobian can be computed."""
+        wires = 4
+        dev = qml.device("default.mixed", wires=wires)
+
+        @qml.qnode(dev, interface="tf", diff_method="backprop")
+        def circuit(param):
+            qml.MultiRZ(param, wires=[0, 1])
+            return qml.probs(wires=list(range(wires)))
+
+        param = tf.Variable(0.3, trainable=True)
+
+        with tf.GradientTape() as tape:
+            out = circuit(param)
+
+        res = tape.gradient(out, param)
+
+        assert np.allclose(res, np.zeros(wires**2))
+
+    def test_full_subsystem(self, mocker):
+        """Test applying a state vector to the full subsystem"""
+        dev = DefaultMixed(wires=["a", "b", "c"])
+        state = tf.constant([1, 0, 0, 0, 1, 0, 1, 1], dtype=tf.complex128) / 2.0
+        state_wires = qml.wires.Wires(["a", "b", "c"])
+
+        spy = mocker.spy(qml.math, "scatter")
+        dev._apply_state_vector(state=state, device_wires=state_wires)
+
+        state = np.outer(state, np.conj(state))
+
+        assert np.all(tf.reshape(dev._state, (-1,)) == tf.reshape(state, (-1,)))
+        spy.assert_not_called()
+
+    def test_partial_subsystem(self, mocker):
+        """Test applying a state vector to a subset of wires of the full subsystem"""
+
+        dev = DefaultMixed(wires=["a", "b", "c"])
+        state = tf.constant([1, 0, 1, 0], dtype=tf.complex128) / np.sqrt(2.0)
+        state_wires = qml.wires.Wires(["a", "c"])
+
+        spy = mocker.spy(qml.math, "scatter")
+        dev._apply_state_vector(state=state, device_wires=state_wires)
+
+        state = np.kron(np.outer(state, np.conj(state)), np.array([[1, 0], [0, 0]]))
+
+        assert np.all(tf.reshape(dev._state, (8, 8)) == state)
+        spy.assert_called()
+
+
+class TestPassthruIntegration:
+    """Tests for integration with the PassthruQNode"""
+
+    def test_jacobian_variable_multiply(self, tol):
+        """Test that jacobian of a QNode with an attached default.qubit.autograd device
```

Comment:
Applies in further cases too.
```suggestion
        """Test that jacobian of a QNode with an attached default.mixed.tf device
```